lib/yarp.rb

```diff
@@ -227,4 +227,84 @@ def self.load(source, serialized)
 require_relative "yarp/ripper_compat"
 require_relative "yarp/serialize"
 require_relative "yarp/pack"
-require "yarp.so"
+
+if RUBY_ENGINE == 'ruby' and false # TODO
+  require "yarp.so"
+else
+  require "yarp.so" # TODO HACK until all methods implemented
+
+  require "fiddle"
+  require "fiddle/import"
+
+  module YARP
+    module LibRubyParser
+      extend Fiddle::Importer
+
+      POINTER_SIZE = sizeof("void*")
+
+      Buffer = struct ['char *value', 'size_t length', 'size_t capacity']
+
+      dlload File.expand_path("../build/librubyparser.so", __dir__)
+
+      typealias 'bool', 'char' # OK? https://github.com/ruby/fiddle/issues/130
```

Comment:
@tenderlove Maybe you know a good way to do this?

Reply:
This is probably fine.  I'll try to fix the issue upstream (tbh I don't really use the C parser functionality in Fiddle very often, and typically manually declare sizes 😅 )

Reply:
kou fixed it already: https://github.com/ruby/fiddle/issues/130
But of course we need this to work on older Fiddle too, so probably this workaround is fine.